pkg/cmd/pulumi/plugin/plugin_run.go

```diff
@@ -100,9 +107,14 @@ func newPluginRunCmd() *cobra.Command {
 				return fmt.Errorf("could not create plugin context: %w", err)
 			}
 
-			plugin, err := plugin.ExecPlugin(pctx, path, pluginDesc, kind, pluginArgs, "", nil, false)
+			plugin, err := plugin.ExecPlugin(pctx, path, source, kind, pluginArgs, "", nil, false)
 			if err != nil {
-				return fmt.Errorf("could not execute plugin %s (%s): %w", pluginDesc, path, err)
+				desc := fmt.Sprintf("%s (%s)", source, path)
+				// don't print the value twice if it's the same
+				if source == path {
+					desc = source
```

Comment:
Personally I'd have reversed this condition so we only need to call `fmt.Sprintf` if it's really needed, but it doesn't matter too much.